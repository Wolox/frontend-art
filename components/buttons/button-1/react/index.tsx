import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
