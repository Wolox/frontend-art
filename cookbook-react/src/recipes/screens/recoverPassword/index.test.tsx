import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import RecoverPassword from '.';

test('renders the component when mounted', () => {
  const { container } = render(<RecoverPassword />);
  expect(container.firstChild).toMatchSnapshot();
});

test('changes email correctly', () => {
  const email = 'example@wolox.com';
  render(<RecoverPassword />);
  const emailInput = screen.getByLabelText('RecoverPassword:email');
  userEvent.type(emailInput, email);

  const submitButton = screen.getByText('RecoverPassword:enter');
  userEvent.click(submitButton);

  // TODO: Expect something when the handleSubmit is implemented
});
