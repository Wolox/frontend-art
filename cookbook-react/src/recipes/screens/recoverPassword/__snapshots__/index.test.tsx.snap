// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the component when mounted 1`] = `
<div
  class="column center full-width container"
>
  <div
    class="column center m-bottom-3"
  >
    <h1
      class="m-bottom-1"
    >
      RecoverPassword:recoverPassword
    </h1>
    <h2>
      RecoverPassword:recoverPasswordDescription
    </h2>
  </div>
  <form
    class="full-width formContainer"
  >
    <div
      class="column start m-bottom-2 sectionContainer"
    >
      <label
        class=" m-bottom-1"
        for="email"
      >
        RecoverPassword:email
      </label>
      <input
        class="m-bottom-2 full-width input input "
        id="email"
        name="email"
        placeholder="RecoverPassword:emailPlaceholder"
        type="text"
      />
      <span
        aria-hidden="true"
        class=" errorText "
        id="email-error"
        role="alert"
      />
    </div>
    <div
      class="column center sectionContainer"
    >
      <button
        class="full-width m-bottom-1 button"
        type="submit"
      >
        RecoverPassword:enter
      </button>
      <a
        href="/login"
      >
        RecoverPassword:returnToLogin
      </a>
    </div>
  </form>
</div>
`;

exports[`renders the component when mounted 1`] = `
<div
  class="column center full-width container"
>
  <div
    class="column center m-bottom-3"
  >
    <h1
      class="m-bottom-1"
    >
      RecoverPassword:recoverPassword
    </h1>
    <h2>
      RecoverPassword:recoverPasswordDescription
    </h2>
  </div>
  <form
    class="full-width formContainer"
  >
    <div
      class="column start m-bottom-2 sectionContainer"
    >
      <label
        class=" m-bottom-1"
        for="email"
      >
        RecoverPassword:email
      </label>
      <input
        class="m-bottom-2 full-width input input "
        id="email"
        name="email"
        placeholder="RecoverPassword:emailPlaceholder"
        type="text"
      />
      <span
        aria-hidden="true"
        class=" errorText "
        id="email-error"
        role="alert"
      />
    </div>
    <div
      class="column center sectionContainer"
    >
      <button
        class="full-width m-bottom-1 button"
        type="submit"
      >
        RecoverPassword:enter
      </button>
      <a
        href="/login"
      >
        RecoverPassword:returnToLogin
      </a>
    </div>
  </form>
</div>
`;
